integration_tests/src/main/python/arithmetic_ops_test.py

```diff
@@ -334,9 +336,13 @@ def test_mod_pmod_by_zero(data_gen, overflow_exp):
 def test_cast_neg_to_decimal_err():
     # -12 cannot be represented as decimal(7,7)
     data_gen = _decimal_gen_7_7
-    exception_content = "Decimal(compact,-120000000,20,0}) cannot be represented as Decimal(7, 7)" \
-        if is_before_spark_314() or ((not is_before_spark_320()) and is_before_spark_322()) else \
-        "Decimal(compact, -120000000, 20, 0) cannot be represented as Decimal(7, 7)"
+    if is_before_spark_314() or ((not is_before_spark_320()) and is_before_spark_322()):
```

Comment:
Just a thought, it would be nice to make this logic consistent across the tests.
e.g. in `test_mod_pmod_by_zero` the logic is a contiguous block but here it's split into `exception_type` + `exception_content`. I think this is probably a good post 11.3 changes issue, this file as a whole is wildly inconsistent across the tests